### PR TITLE
Prevent Lottie from running in the background after navigating to other pages

### PR DESCRIPTION
**Pull request by QichenZhu**
### Fixed Issues

$ https://github.com/Expensify/App/issues/47273
PROPOSAL: https://github.com/Expensify/App/issues/47273#issuecomment-2294733914

### Tests

1. Go to Settings - Preferences.
2. Navigate to Chat.
3. Navigate back to the Settings tab.
4. Click on Priority Mode, Language, or Theme.

**Expected result:** The animation in the Settings tab should work, and the right-side modal should open.

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Go to Settings - Preferences.
2. Navigate to Chat.
3. Navigate back to the Settings tab.
4. Click on Priority Mode, Language, or Theme.

**Expected result:** The animation in the Settings tab should work, and the right-side modal should open.

### QA Steps

1. Go to Settings - Preferences.
2. Navigate to Chat.
3. Navigate back to the Settings tab.
4. Click on Priority Mode, Language, or Theme.

**Expected result:** The animation in the Settings tab should work, and the right-side modal should open.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/88b5336c-deac-44f0-8d37-de2420c79b5a

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/b13144ac-2eac-4992-87df-aa528248da95

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/2752b1e4-c8b4-401e-bdc7-bd6676fd2e48

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/91097b86-ca0b-43e5-9942-4aa1344d831f

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/0892807b-2159-4e09-b663-40d9520cf306

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/1192d734-caf9-40a8-8120-281d13b31fa8

</details>

### CODE REVIEW

File: src/components/Lottie/index.tsx [hunk 1]
```diff
@@ -27,13 +31,33 @@ function Lottie({source, webStyle, ...props}: Props, ref: ForwardedRef<LottieVie
         setAnimationFile(source.file);
     }, [setAnimationFile, source.file]);
 
+    useEffect(() => {
+        const unsubscribe = navigation.addListener('focus', () => {
+            setIsHidden(false);
+        });
+        return unsubscribe;
```
Comment (QichenZhu) [inline src/components/Lottie/index.tsx:38]:
```suggestion
        return unsubscribeNavigationFocus;
```

File: src/components/Lottie/index.tsx [hunk 2]
```diff
@@ -27,13 +31,33 @@ function Lottie({source, webStyle, ...props}: Props, ref: ForwardedRef<LottieVie
...
+    // See https://github.com/Expensify/App/issues/47273
+    useEffect(() => {
+        const unsubscribe = navigation.addListener('blur', () => {
+            const state = navigation.getState();
+            const targetRouteName = state?.routes?.[state?.index ?? 0]?.name;
+            if (targetRouteName !== NAVIGATORS.RIGHT_MODAL_NAVIGATOR) {
+                setIsHidden(true);
+            }
+        });
+        return unsubscribe;
```
Comment (QichenZhu) [inline src/components/Lottie/index.tsx:51]:
```suggestion
        return unsubscribeNavigationBlur;
```

**Review by suneox — APPROVED** [review]
LGTM

**Comment by QichenZhu** [comment]
@suneox @aldo-expensify Could we pause the review for now? RN was upgraded to 0.75 less than an hour ago, which is a major change. I need to retest on all platforms tomorrow to be safe.

**Comment by QichenZhu** [comment]
@suneox @aldo-expensify Retested and found no major differences after merging the `main` branch, so it’s ready for review.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/user-attachments/assets/14a56cb2-b6c0-4bea-88ca-6a353e01adf4


</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/user-attachments/assets/6adf861c-635a-41c9-9786-7b55ed877e64


</details>

<details>
<summary>iOS: Native</summary>



https://github.com/user-attachments/assets/b9b3781f-ae76-46d5-aa34-b28c7ce3043d


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/user-attachments/assets/ae2e827d-3ca3-4fda-90a9-00e05055233c


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/user-attachments/assets/4d127890-1d18-4021-a676-15251d98ba39


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/user-attachments/assets/d259e7e0-6e55-4c24-b553-f16baffccb90


</details>

**Review by aldo-expensify — APPROVED** [review]

**Comment by MonilBhavsar** [comment]
We reverted this PR because it caused a deploy blocker that caused an App crash https://github.com/Expensify/App/pull/48283

**Comment by QichenZhu** [comment]
@suneox Oops, my bad. Didn't realize `useNavigation()` could throw an error if the component is ouside of `NavigationContainer`.

**Comment by QichenZhu** [comment]
@suneox Yes, I can still reproduce it.

Screenshot 2024-09-03 at 2 01 09 AM

File: src/components/Lottie/index.tsx [hunk 3]
```diff
@@ -27,13 +31,33 @@ function Lottie({source, webStyle, ...props}: Props, ref: ForwardedRef<LottieVie
...
+    }, [navigation]);
+
+    // Prevent the animation from running in the background after navigating to other pages.
+    // See https://github.com/Expensify/App/issues/47273
+    useEffect(() => {
+        const unsubscribeNavigationBlur = navigation.addListener('blur', () => {
+            const state = navigation.getState();
+            const targetRouteName = state?.routes?.[state?.index ?? 0]?.name;
+            if (targetRouteName !== NAVIGATORS.RIGHT_MODAL_NAVIGATOR) {
+                setIsHidden(true);
```
Comment (suneox) [inline src/components/Lottie/index.tsx:48]: This change causes an issue where assets are hidden during the screen transition, and in some cases, if the transition is slow, users may briefly see a blank view.​

**Pull request merged**